ui/components/app/permission-connect-header/permission-connect-header.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  AlignItems,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+  Display,
+  BlockSize,
+  FontWeight,
+  FlexDirection,
+  BackgroundColor,
+} from '../../../helpers/constants/design-system';
+import {
+  IconSize,
+  Text,
+  Box,
+  AvatarFavicon,
+  AvatarBase,
+} from '../../component-library';
+
+const PermissionConnectHeader = ({ origin, iconUrl }) => {
+  const transformOriginToTitle = (rawOrigin) => {
+    try {
+      const url = new URL(rawOrigin);
+      const parts = url.hostname.split('.');
+      return parts.slice(-2).join('.');
+    } catch (e) {
+      return 'Unknown Origin';
+    }
+  };
+  const title = transformOriginToTitle(origin);
+
+  return (
+    <Box
+      backgroundColor={BackgroundColor.backgroundDefault}
+      width={BlockSize.Full}
+      alignItems={AlignItems.center}
+      display={Display.Flex}
+      padding={4}
+      style={{
+        boxShadow: 'var(--shadow-size-lg) var(--color-shadow-default)',
+      }}
+    >
+      <Box>
+        {iconUrl ? (
+          <AvatarFavicon
+            backgroundColor={BackgroundColor.backgroundAlternative}
+            size={IconSize.Lg}
+            src={iconUrl}
+            name={title}
+          />
+        ) : (
+          <AvatarBase
+            size={IconSize.Lg}
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.center}
+            color={TextColor.textAlternative}
+            style={{ borderWidth: '0px' }}
+            backgroundColor={BackgroundColor.backgroundAlternative}
+          >
+            {title?.match(/[a-z0-9]/iu)?.[0] ?? '?'}
```

Comment:
I see this regex is in `ui/components/app/snaps/snap-avatar/snap-avatar.js` as well; I feel like at the least, we should make this a helper function, or should the `SnapAvatar` component be used here?

Reply:
Refactored now. I extracted the regex things into a helper/utils function.

We didn't want to use Snap components for the native connection/permission flows. I started doing it first, but it just bloated the Snaps components code and made it more difficult to understand and maintain. Snaps components have diffrent requirements, so I had to add too many conditions/branches in order to make it.

I would prefer keeping these things separate and following their own directions in future development.